syntax = "proto3";

package marty.pkd.v1;

import "google/protobuf/empty.proto";

message TrustAnchor {
  string certificate_id = 1;
  string subject = 2;
  string certificate_pem = 3;
  string storage_key = 4;
  string not_after = 5;
  bool revoked = 6;
}

message ListTrustAnchorsResponse {
  repeated TrustAnchor anchors = 1;
}

message SyncRequest {
  bool force_refresh = 1;
}

message SyncResponse {
  bool success = 1;
  string message = 2;
}

service PKDService {
  rpc ListTrustAnchors(google.protobuf.Empty) returns (ListTrustAnchorsResponse);
  rpc Sync(SyncRequest) returns (SyncResponse);
}
